### PR TITLE
data importing and EDA

### PR DESCRIPTION
**Pull request by fiaaatlux**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
+  - jupyterlab=4.2.6
+  - jupyterlab_server=2.27.3
+  - jupyter_server=2.14.2
+  - jupyter_core=5.7.2
+  - jupyter_client=8.6.3
+  - ipykernel=6.29.5
+  - nb_conda_kernels=2.5.1
+  - matplotlib=3.9.2
+  - scipy=1.14.1
+  - vega_datasets
```
Comment (Psingh238) [inline environment.yml:23]: Remove `vega_datasets` since we do not need datasets from this package.

File: environment.yml [hunk 2]
```diff
@@ -31,17 +40,18 @@ dependencies:
...
-      - datasets
-      - ultralytics
-      - psutil>=5.7.2
+    - altair-ally>=0.1.1
+    - ucimlrepo==0.0.7
+    - vegafusion==1.6.9
+    - vegafusion-python-embed==1.6.9
+    - mglearn
+    - spacymoji
+    - otter-grader>=6.0
```
Comment (Psingh238) [inline environment.yml:53]: Do not need `otter-grader` since it is unused.

File: environment.yml [hunk 3]
```diff
@@ -1,28 +1,37 @@
...
+  - eli5
+  - shap
   - jinja2
   - lightgbm
+  - spacy
+  - xgboost
+  - catboost
+  - nltk
+  - imbalanced-learn
   - torchvision
```
Comment (Psingh238) [inline environment.yml:35]: Remove pytorch related packages since it will be unused.

File: environment.yml [hunk 4]
```diff
@@ -1,28 +1,37 @@
-# Install the course conda environment with the following command:
-# conda env create -f env-dsci-571.yml
-# Once you install it, you should be able to see it in your Jupyter Lab.
-# If not, check the documentation here: https://ipython.readthedocs.io/en/stable/install/kernel_install.html
-name: "522"
-
+name: '522'
 channels:
   - conda-forge
   - pytorch
```
Comment (Psingh238) [inline environment.yml:4]: Remove pytorch channel since pytorch packages are removed.

File: wine_quality.ipynb
Comment (Psingh238) [inline wine_quality.ipynb:1]: For the second code cell, consider adding the following to save the data to the data folder as well:
```suggestion
raw_data.to_csv('../data/raw/wine_quality.csv', index=False)
```
This fulfills the requirement to also load data and have it written to the data folder through the code instead of downloading the data manually.

**Review by Psingh238 — CHANGES_REQUESTED** [review]
Suggested changes for the environment.yml file and the analysis code.

File: environment.yml (hunk 1 above)
Comment (fiaaatlux) [reply environment.yml:23]: ok

File: environment.yml (hunk 2 above)
Comment (fiaaatlux) [reply environment.yml:53]: ok

File: environment.yml (hunk 3 above)
Comment (fiaaatlux) [reply environment.yml:35]: I suggest to keep it for now since we not sure if we gna use it later for the model

File: environment.yml (hunk 4 above)
Comment (fiaaatlux) [reply environment.yml:4]: I suggest to keep it for now since we not sure if we gna use it later for the model

Comment (Psingh238) [reply environment.yml:4]: Alright, understood.

File: environment.yml (hunk 3 above)
Comment (Psingh238) [reply environment.yml:35]: Understood, we will keep it for now.

File: wine_quality.ipynb
Comment (Psingh238) [reply wine_quality.ipynb:1]: Updated suggestion to put raw data in the raw subfolder while processed data can be stored in a processed subfolder.

Comment (Psingh238) [inline wine_quality.ipynb:1]: Suggested to move the notebook to a `notebooks` folder. Check current `main` branch for organization.

**Review by Psingh238 — CHANGES_REQUESTED** [review]
Suggested changes for file organization. Please make the necessary changes to comply with optimal file structures.
